cluster/store/meta_class.go

```diff
@@ -0,0 +1,217 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package store
+
+import (
+	"fmt"
+	"sync"
+
+	command "github.com/weaviate/weaviate/cluster/proto/cluster"
+	"github.com/weaviate/weaviate/entities/models"
+	"github.com/weaviate/weaviate/usecases/sharding"
+	"golang.org/x/exp/slices"
+)
+
+type metaClass struct {
+	sync.RWMutex
+	Class    models.Class
```

Comment:
**thought (non-blocking):** i there is a reason for exporting it ?

Reply:
This is not exported because metaClass is private

Reply:
sure, that what I meant so i think then we don't need to export `Class` and `Sharding` ?

Reply:
Could you explain what do you mean by **export Class and Sharding**.
 Both properties are not exported outside the package because metadata is private

Reply:
yes i agree, i just meant by converting it to `class` and `sharding`, so that it's following the convention though 

```
type metaClass struct {
	sync.RWMutex
	class    models.Class
        sharding sharding.State
}
```

Reply:
I'm not familiar with this convention, but I prefer the current version as it is more readable to me